### DISCUSSION TITLE
Mydataset is not in the dataset registry

### DISCUSSION THREAD

**Discussion by yourbikun** [open-mmlab/mmdetection#9395] [Q&A] [answered]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I have modified the scripts/configs, or I'm working on my own tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection

### Environment

TorchVision: 0.10.1+cu111
OpenCV: 3.4.10
MMCV: 1.6.2
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 11.1
MMDetection: 2.25.2+

### Reproduces the problem - code sample

Mydataset is not in the dataset registry

### Reproduces the problem - command or script

python tools/train.py configs/rpc/faster_rcnn_r50_fpn_1x_mydata.py

### Reproduces the problem - error message

Mydataset is not in the dataset registry

### Additional information

I found when I run ‘import mmdet ’ and ‘print(mmdet.path)’，get the results of ['/usr/local/lib/python3.8/dist-packages/mmdet'].
So When I run the train file, Got the error 'Mydataset is not in the dataset registry'.
Excuse me, how can I locate my mmdet to my source code
thank you for your help

**Comment by ZwwWayne** [comment, marked as answer]
You should make sure that the file you added is imported. Please follow https://mmdetection.readthedocs.io/en/latest/tutorials/customize_models.html#import-the-module to use custom_imports or update __init__.py to import your module.

**Comment by yourbikun** [comment]
I have imported the file I added. 
The main problem is that the system imports mmdet from pip, not from the source code I modified.
I solved this problem in this way:
 First, pip uninstall mmdet
Second, In tool/train Add "import sys" at the top of the py file
import sys
o_ path = os.getcwd()”
Sys. path. append (o_path), so that the system can import the mmdet folder I modified.
But I don't know how other people solve this problem, and I also want to listen to other people's suggestions

**Comment by yourbikun** [comment]
ain problem is that the system imports mmdet from pip, not from the source code I modified.
I solved this problem in this way:
 First, pip uninstall mmdet
Second, In tool/train Add "import sys" at the top of the py file
import sys
o_ path = os.getcwd()”
Sys. path. append (o_path), so that the system can import the mmdet folder I modified.
But I don't know how other people solve this problem, and I also want to listen to other people's suggestions
